In [1]:
# Import Dependencies
import pandas as pd

In [2]:
# Use pgAdmin to export the vine_table as a csv file
# this file has been saved as vine_table.csv in this repository

# Read in the vine_table.csv file
vine_df = pd.read_csv("./Data/vine_table.csv")
vine_df

review_id  star_rating  helpful_votes  total_votes vine  \
0        R1WBPB8MDCCN8F          5.0            0.0          0.0    N   
1        R32M0YEWV77XG8          5.0            1.0          1.0    N   
2         RR8V7WR27NXJ5          1.0            0.0          0.0    N   
3        R1MHO5V9Z932AY          5.0            0.0          0.0    N   
4        R16PD71086BD2V          5.0            0.0          1.0    N   
...                 ...          ...            ...          ...  ...   
4850355  R1VERHFY08SLZS          1.0            6.0          8.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   
4850357   R2YECHGBV291H          5.0            1.0          1.0    N   
4850358  R2Z784CWN7JZ4X          2.0            3.0          4.0    N   
4850359  R2J13W5P3N2KV4          4.0           18.0         19.0    N   

        verified_purchase  
0                       N  
1                       Y  
2                       Y  
3                       Y  
4                       N  
...                   ...  
4850355                 N  
4850356                 N  
4850357                 N  
4850358                 N  
4850359                 N  

[4850360 rows x 6 columns]

In [3]:
# explore the dataframe using the info method
vine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4850360 entries, 0 to 4850359
Data columns (total 6 columns):
 #   Column             Dtype  
---  ------             -----  
 0   review_id          object 
 1   star_rating        float64
 2   helpful_votes      float64
 3   total_votes        float64
 4   vine               object 
 5   verified_purchase  object 
dtypes: float64(3), object(3)
memory usage: 222.0+ MB


## Step 1 ##

Filter the data and create a new DataFrame or table to retrieve all the rows where the total_votes count is equal to or greater tahn 20 to pick reviews that are more likely to be helpful and to avoid having division by zero erros later on.

In [4]:
# filter vine_df for all rowas where total_votes >= 20

vine_df_20 = vine_df.loc[vine_df["total_votes"] >= 20]
vine_df_20

# Note this reduces the total rows from 4,850,360 to 67,855

review_id  star_rating  helpful_votes  total_votes vine  \
326      R1B4753QMTD8ON          5.0           32.0         32.0    N   
662      R28BVHOEL6GCHZ          5.0           21.0         21.0    N   
1055     R27IB4BRKO71I0          2.0           16.0         20.0    N   
1225      RJ86OW12Y4NRL          5.0          124.0        128.0    N   
1448     R3KRABOWCH6ZE4          1.0           48.0         65.0    N   
...                 ...          ...            ...          ...  ...   
4850347  R1KGFSPKL3YCGK          4.0           25.0         26.0    N   
4850349  R15AZWCW5ZSFSN          5.0          157.0        160.0    N   
4850352  R1ZWVHA89AE49B          5.0           32.0         33.0    N   
4850353  R115HJRM6FUVIC          5.0           27.0         27.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   

        verified_purchase  
326                     N  
662                     Y  
1055                    Y  
1225                    Y  
1448                    Y  
...                   ...  
4850347                 N  
4850349                 Y  
4850352                 N  
4850353                 N  
4850356                 N  

[67855 rows x 6 columns]

## Step 2 ##

Filter the new DataFrame or table created in Step 1 and create a new DataFrame or table to retrieve all the rows where the number of helpful_votes divided by total_votes is equal to or greater tahn 50%.

In [5]:
# filter vine_df_20 where helpful_votes / total_votes >= 0.50

vine_df_50 = vine_df_20.loc[(vine_df_20["helpful_votes"] / vine_df_20["total_votes"]) >= 0.50]
vine_df_50

# this reduces the number of rows form 67,855 to 61,948

review_id  star_rating  helpful_votes  total_votes vine  \
326      R1B4753QMTD8ON          5.0           32.0         32.0    N   
662      R28BVHOEL6GCHZ          5.0           21.0         21.0    N   
1055     R27IB4BRKO71I0          2.0           16.0         20.0    N   
1225      RJ86OW12Y4NRL          5.0          124.0        128.0    N   
1448     R3KRABOWCH6ZE4          1.0           48.0         65.0    N   
...                 ...          ...            ...          ...  ...   
4850347  R1KGFSPKL3YCGK          4.0           25.0         26.0    N   
4850349  R15AZWCW5ZSFSN          5.0          157.0        160.0    N   
4850352  R1ZWVHA89AE49B          5.0           32.0         33.0    N   
4850353  R115HJRM6FUVIC          5.0           27.0         27.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   

        verified_purchase  
326                     N  
662                     Y  
1055                    Y  
1225                    Y  
1448                    Y  
...                   ...  
4850347                 N  
4850349                 Y  
4850352                 N  
4850353                 N  
4850356                 N  

[61948 rows x 6 columns]

## Step 3 ##

Filter the DataFrame or table created in Step 2, and create a new DataFrame or table that retrieves all the rows where a review was written as part of the Vine program (paid), vine =='Y'.

In [6]:
# filter vine_df_50 where vine = 'Y'

paid_reviews_df = vine_df_50.loc[vine_df_50["vine"] == 'Y']
paid_reviews_df

# note there are 334 paid reviews (that meet the filter criteria from steps 1 & 2).

review_id  star_rating  helpful_votes  total_votes vine  \
5506     R32OMS1LSDRCOB          4.0           27.0         31.0    Y   
24578    R2IPKK4I0XRCFR          4.0           16.0         21.0    Y   
58364     RXYYJ8SGI9IX6          5.0           36.0         41.0    Y   
63194    R1SAUBFC7T6JHC          5.0           73.0         75.0    Y   
177724   R329HVJKKHWSG3          5.0           59.0         59.0    Y   
...                 ...          ...            ...          ...  ...   
4717565  R1WG2F3DZFE301          5.0           51.0         81.0    Y   
4719313  R3VB53V1VDWPQ2          4.0          104.0        132.0    Y   
4754150   RCKH2IYVH9PCG          4.0           21.0         21.0    Y   
4770385  R1OR7JY4MQP6I7          5.0          181.0        188.0    Y   
4770596  R34IO2BR3JWIIZ          4.0           17.0         20.0    Y   

        verified_purchase  
5506                    N  
24578                   N  
58364                   N  
63194                   N  
177724                  N  
...                   ...  
4717565                 N  
4719313                 N  
4754150                 N  
4770385                 N  
4770596                 N  

[334 rows x 6 columns]

## Step 4 ##

In [8]:
# filter vine_df_50 where vine = "N"
unpaid_reviews_df = vine_df_50.loc[vine_df_50["vine"] == 'N']
unpaid_reviews_df
                                   
# note there are 61614 unpaid reviews (that meet the criteria from steps 1 & 2)

review_id  star_rating  helpful_votes  total_votes vine  \
326      R1B4753QMTD8ON          5.0           32.0         32.0    N   
662      R28BVHOEL6GCHZ          5.0           21.0         21.0    N   
1055     R27IB4BRKO71I0          2.0           16.0         20.0    N   
1225      RJ86OW12Y4NRL          5.0          124.0        128.0    N   
1448     R3KRABOWCH6ZE4          1.0           48.0         65.0    N   
...                 ...          ...            ...          ...  ...   
4850347  R1KGFSPKL3YCGK          4.0           25.0         26.0    N   
4850349  R15AZWCW5ZSFSN          5.0          157.0        160.0    N   
4850352  R1ZWVHA89AE49B          5.0           32.0         33.0    N   
4850353  R115HJRM6FUVIC          5.0           27.0         27.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   

        verified_purchase  
326                     N  
662                     Y  
1055                    Y  
1225                    Y  
1448                    Y  
...                   ...  
4850347                 N  
4850349                 Y  
4850352                 N  
4850353                 N  
4850356                 N  

[61614 rows x 6 columns]

## Step 5 ##

Determine the total number of reviews, the number of 5-star reviews, and the percentage of 5-star reviews for the two types of review (paid vs unpaid).

## Step 5a. Paid Reviews ##

In [14]:
# Determine the count of paid 5 star reviews
paid_5star_reviews = paid_reviews_df.loc[paid_reviews_df["star_rating"] == 5]["star_rating"].count()
paid_5star_reviews

139

In [15]:
# Determine the count of paid reviews
paid_reviews = paid_reviews_df["star_rating"].count()
paid_reviews

334

In [16]:
# Determine the percentage of paid_5star_reviews
# percentage_paid_5star_reviews = paid_5star_reviews / paid_reviews
percentage_paid_5star_reviews = round((paid_5star_reviews / paid_reviews) *100, 2)
percentage_paid_5star_reviews


41.62

## Step 5b Unpaid Reviews ##

In [17]:
# Determine the count of unpaid 5 star reviews
unpaid_5star_reviews = unpaid_reviews_df.loc[unpaid_reviews_df["star_rating"] == 5]["star_rating"].count()
unpaid_5star_reviews

32665

In [20]:
# Determine the count of unpaid reviews
unpaid_reviews = unpaid_reviews_df["star_rating"].count()
unpaid_reviews

61614

In [21]:
# Determine the percentage of unpaid_5star_reviews
# percentage_paid_5star_reviews = paid_5star_reviews / paid_reviews
percentage_unpaid_5star_reviews = round((unpaid_5star_reviews / unpaid_reviews) *100, 2)
percentage_unpaid_5star_reviews


53.02

In [23]:
## Analysis: ##

Ho: 5star_vine = 5star_not_vine (no difference in the % of 5 star ratings paid vs unpaid)

Ha: 5star_vine <> 5star_not_vine (there is a difference)

    - percentage_5star_vine = 41.62%
                                 
    - percentage_5star_not_vine = 53.02%
    
    - Finding: The percentage of 5 star ratings is not the same when comparing vine reviews to not vine reviews.
    
    - There is no reason to believe that the percentage of 5 star reviews is not directly tied to whether the reviewer is paid or not.
    
    - A hypothesis test should be run to evaluate the proportion of 5 star reviews among the samples
    
    